src/Containers/OhmmsPETE/OhmmsArray.h

```diff
@@ -95,6 +97,74 @@ class Array
     return X.device_data();
   }
 
+  template<typename SIZET = size_t, typename = std::is_integral<SIZET>>
+  Type_t* data(const std::array<SIZET, D>& offsets)
+  {
+    size_t offset = offsets[0];
+    for (int i = 1; i < offsets.size(); i++)
+      offset = offset * Length[i] + offsets[i];
+    return X.data() + offset;
+  }
+  template<typename SIZET = size_t, typename = std::is_integral<SIZET>>
+  const Type_t* data(const std::array<SIZET, D>& offsets) const
+  {
+    size_t offset = offsets[0];
+    for (int i = 1; i < offsets.size(); i++)
+      offset = offset * Length[i] + offsets[i];
+    return X.data() + offset;
+  }
+  template<typename SIZET     = size_t,
+           typename           = std::is_integral<SIZET>,
+           typename Allocator = ALLOC,
+           typename           = qmcplusplus::IsDualSpace<Allocator>>
+  Type_t* device_data(const std::array<SIZET, D>& offsets)
+  {
+    size_t offset = offsets[0];
+    for (int i = 1; i < offsets.size(); i++)
+      offset = offset * Length[i] + offsets[i];
+    return X.device_data() + offset;
+  }
+  template<typename SIZET     = size_t,
+           typename           = std::is_integral<SIZET>,
+           typename Allocator = ALLOC,
+           typename           = qmcplusplus::IsDualSpace<Allocator>>
+  const Type_t* device_data(const std::array<SIZET, D>& offsets) const
+  {
+    size_t offset = offsets[0];
+    for (int i = 1; i < offsets.size(); i++)
+      offset = offset * Length[i] + offsets[i];
+    return X.device_data() + offset;
+  }
+
+  /// access data pointer at (offset_1, ..., offset_D})
+  template<typename... Args>
+  Type_t* data(Args... offsets)
+  {
+    static_assert(sizeof...(Args) == D, "resize arguments must match dimensionality of Array");
```

Comment:
I think the assert message needs to be updated.

Reply:
Fixed.